### DISCUSSION TITLE
Retrieving Private Key in Unity

### DISCUSSION THREAD

**Discussion by voodlaiz** [Web3Auth/Web3Auth#549] [Help]
Hello, I'm trying to retrieve the private key of the user and create a new user in my game using that PK. However, if I create a player that has connected with Google and afterwards, he connects to Web3Auth with, let's say, Discord, the PK is different. How can I retrieve the unique PK of the user? (Hope my question is clear, thanks)

**Comment by yashovardhan** [comment]
Hey @voodlaiz, 

Thanks for your question.

Talking about your particular question, every private key is uniquely generated for each combination of a `verifier` and `clientId`. Each social provider has a different verifier, which means that the private key will be different for each social provider. Hence, this is causing your Discord PKs to be different from Google PKs.

Coming to a solution for this problem, you can ask our team to create an [aggregate verifier](https://web3auth.io/docs/custom-authentication/verifiers#aggregate-verifiers) for you, where you can provide multiple different login providers and a common fielding linking them. This will enable web3auth to generate the same key for both the social providers. Currently there is no self serve way of creating a aggregate verifier. You can to [get in touch](https://web3auth.io/contact-us.html) with our team and they can help you create one.

**Reply by voodlaiz** [reply]
Thanks for the clear answer, I'll dig into the aggregate verifier to see if we can come up with a solution.
